File: Dagger2Ex/app/src/main/java/expencive/vk/com/dagger2ex/dagger/PetroilEngineModule.java
package expencive.vk.com.dagger2ex.dagger;


import dagger.Binds;
import dagger.Module;
import expencive.vk.com.dagger2ex.car.Engine;
import expencive.vk.com.dagger2ex.car.PetroilEngine;

@Module
public abstract class PetroilEngineModule {

    @Binds
    abstract Engine bindEngine(PetroilEngine engine);

}

File: DaggerKotlin/app/src/main/java/vk/expencive/daggerkotlin/dagger/PetroilEngineModule.kt
package vk.expencive.daggerkotlin.dagger

import dagger.Binds
import dagger.Module
import vk.expencive.daggerkotlin.car.Engine
import vk.expencive.daggerkotlin.car.PetroilEngine


@Module
abstract class PetroilEngineModule {

    @Binds
    abstract fun bindPetroilEngine(engine: PetroilEngine): Engine
}